### PR TITLE
Add platform macros

### PR DESCRIPTION
**Pull request by adamscott**
Add platform macros, like `LINUX_ENABLED` and `UNIX_ENABLED` on Linux, `MACOS_ENABLED`, `UNIX_ENABLED`, ~~`COREAUDIO_ENABLED`, `COREMIDI_ENABLED`~~ on macos, and `WINDOWS_ENABLED`, ~~`WASAPI_ENABLED`, `WINMIDI_ENABLED`~~ on Windows.

### CODE REVIEW

File: SConstruct [hunk 1]
```diff
@@ -192,6 +192,14 @@ target_tool.options(opts)
 opts.Update(env)
 Help(opts.GenerateHelpText(env))
 
+# Process platform argument
+if env.get("platform", default_platform) == "linux":
+    env.Append(CPPDEFINES=["LINUX_ENABLED", "UNIX_ENABLED"])
```
Comment (adamscott) [inline SConstruct:197]: Is there a reason that GDExtension doesn't build the "linuxbsd" platform instead of just "linux"?

**Comment by dsnopek** [comment]
Thanks! I haven't had a chance to do a proper review yet, but I think this is a great idea

**Review by Faless — CHANGES_REQUESTED** [review]
Please move the flags to the dedicated platform tools in `tools`, inside the `generate` function.
These are the "equivalent" of upstream Godot `configure` functions inside `platform/name/detect.py`:
- `tools/windows.py`
- `tools/linux.py`
- `tools/macos.py`
- `tools/ios.py`
- etc.

This will also simplify the code as you can just add the flags there without having to check for the `env["platform"]` variables (the `generate` function is called only when the given platform is selected).

**Comment by adamscott** [comment]
> Please move the flags to the dedicated platform tools in `tools`, inside the `generate` function.

Yes, I saw this yesterday, but didn't have time to change it. Thanks, will do.

File: tools/ios_osxcross.py [hunk 2]
```diff
@@ -24,3 +24,5 @@ def generate(env):
         ]
     )
     env.Append(CCFLAGS=["-stdlib=libc++"])
+
+    env.Append(CPPDEFINES=["IOS_ENABLED", "UNIX_ENABLED", "COREAUDIO_ENABLED"])
```
Comment (Faless) [inline tools/ios_osxcross.py:28]: This is not needed, as this tool is included by `ios.py` (I think we should actually move the code in this tool to be the same `ios.py` file).
```suggestion
```

File: tools/macos_osxcross.py [hunk 3]
```diff
@@ -26,3 +26,5 @@ def generate(env):
     if binpath not in env["ENV"]["PATH"]:
         # Add OSXCROSS bin folder to PATH (required for linking).
         env["ENV"]["PATH"] = "%s:%s" % (binpath, env["ENV"]["PATH"])
+
+    env.Append(CPPDEFINES=["MACOS_ENABLED", "UNIX_ENABLED", "COREAUDIO_ENABLED", "COREMIDI_ENABLED"])
```
Comment (Faless) [inline tools/macos_osxcross.py:30]: Like `ios_oscross.py`, this is also included by `macos.py`,

```suggestion
```

**Review by Faless — CHANGES_REQUESTED** [review]
See my 2 comments, I will open a separate PR to merge the `osxcross` tools and reduce confusion.

**Review by Faless — APPROVED** [review]
This looks good now :+1: .
The only thing I'm not sure about is having `COREAUDIO_ENABLED`, `COREMIDI_ENABLED` on ios/macos, and `WASAPI_ENABLED`, `WINMIDI_ENABLED` on windows.

Why are they useful? I think those are there to control godot driver internals, and if they can't be disabled on those platforms, then extensions could just check for `WINDOWS_ENABLED`/`MACOS_ENABLED`.

**Comment by adamscott** [comment]
@Faless I just removed the non-platform related macros!

**Review by dsnopek — APPROVED** [review]
Thanks!

This looks great to me. I did a quick test, and I can see the Linux defines showing up as expected.

**Pull request merged**